### PR TITLE
Redis: Redis setup

### PR DESCRIPTION
**Pull request by MHA10**
- Redis setup
- Redis Server health check

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+PORT=3000
+
+#gRPC
+GRPC_SERVER_HOST=localhost
+GRPC_SERVER_PORT=5000
+
+#redis
+REDIS_HOST=localhost
+REDIS_PORT=6379
+REDIS_PASSWORD=Abcd@1234
```
Comment (MHassanTariq) [inline .env.example:21]: Is this intentional?

File: src/config/app.config.module.ts [hunk 2]
```diff
@@ -3,7 +3,9 @@ import { TypeOrmModule } from '@nestjs/typeorm';
 import databaseConfig from './ormconfig';
 import { ServerHealthCheckModule } from 'src/modules/server-health-check/server-health-check.module';
 import { RouterModule } from '@nestjs/core';
+
 import { AuthModule } from 'src/modules/auth/auth.module';
+import { NestConfigModule } from './nest.config.module';
```
Comment (MHassanTariq) [inline src/config/app.config.module.ts:8]: Is this import used in the code?

File: src/config/gRPC/grpc.config.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { join } from 'path';
+import { ClientOptions, Transport } from '@nestjs/microservices';
+
+export const grpcConfig: ClientOptions = {
+  transport: Transport.GRPC,
+  options: {
+    package: 'auth',
+    protoPath: join(__dirname, '../../../src/modules/auth/dto/auth.proto'),
```
Comment (MHassanTariq) [inline src/config/gRPC/grpc.config.ts:8]: You mentioned that path will be decided on the modular level. Is this intentional?

File: src/config/swagger/swagger.config.ts [hunk 4]
```diff
@@ -9,6 +9,4 @@ export const swaggerConfig = new DocumentBuilder()
     { type: 'apiKey', name: 'x-api-key', in: 'header' },
     'api-gateway-key',
   )
-  .addTag('Health Check')
```
Comment (MHassanTariq) [inline src/config/swagger/swagger.config.ts:12]: Did you try test Swagger documentation page after removing these tags? Does it create any impact on the api docs?

File: src/modules/auth/controllers/auth.controller.ts [hunk 5]
```diff
@@ -1,20 +1,34 @@
...
+  @ApiResponse({ status: 201, description: 'Success.' })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  async signup(@Body() signupInput: SignUpDto): Promise<boolean> {
+    try {
+      const signupResponse = await lastValueFrom(
+        this.grpcAuthClientService.signup(signupInput),
+      );
+      return signupResponse;
+    } catch (error) {
+      console.error('🚀 ~ HeroesController ~ createHero ~ error:', error);
```
Comment (MHassanTariq) [inline src/modules/auth/controllers/auth.controller.ts:30]: Unintentional Logger committed.

File: src/modules/auth/dto/auth.dto.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty, IsString } from 'class-validator';
+
+export class SignUpDto {
```
Comment (MHassanTariq) [inline src/modules/auth/dto/auth.dto.ts:4]: You will need to share the organisation name / id, and admin uuid incase you want to signup a new user.

File: src/modules/server-health-check/controllers/redis-health-check.controller.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { Controller, Get, InternalServerErrorException } from '@nestjs/common';
```
Comment (MHassanTariq) [inline src/modules/server-health-check/controllers/redis-health-check.controller.ts:1]: I think one other approach could be exposing a single `/health-check` API to fetch the health status of all integrated services. 
This approach might require building multiple APIs for different integrated services.

**Review by MHassanTariq — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (MHA10) [reply .env.example:21]: Yes just to give the idea for dev purposes, what are the expected values.

File: src/config/app.config.module.ts (hunk 2 above)
Comment (MHA10) [reply src/config/app.config.module.ts:8]: Nope, need to remove it.

File: src/config/gRPC/grpc.config.ts (hunk 3 above)
Comment (MHA10) [reply src/config/gRPC/grpc.config.ts:8]: Yup, this is intentional, and just for now.

File: src/config/swagger/swagger.config.ts (hunk 4 above)
Comment (MHA10) [reply src/config/swagger/swagger.config.ts:12]: Yup it worked fine without these.

File: src/modules/auth/controllers/auth.controller.ts (hunk 5 above)
Comment (MHA10) [reply src/modules/auth/controllers/auth.controller.ts:30]: Logging the error.

File: src/modules/auth/dto/auth.dto.ts (hunk 6 above)
Comment (MHA10) [reply src/modules/auth/dto/auth.dto.ts:4]: Yup, will do when implementing that side.

File: src/modules/server-health-check/controllers/redis-health-check.controller.ts (hunk 7 above)
Comment (MHA10) [reply src/modules/server-health-check/controllers/redis-health-check.controller.ts:1]: Yup we can, will make it as one.

**Review by MHassanTariq — APPROVED** [review]

**Pull request merged**